tools/api-markdown-documenter/src/documentation-domain/DocumentNode.ts

```diff
@@ -38,6 +38,31 @@ export interface DocumentNodeProperties {
 	readonly frontMatter?: string;
 }
 
+/**
+ * Metadata of a {@link DocumentNode} in terms of its API.
+ *
+ * @remarks
+ * `DocumentItemMetadata` aids in tracing a documentation node to its API, useful for cross-referencing and integrations.
+ *
+ * @public
+ */
+export interface DocumentItemMetadata {
+	/**
+	 * Name of the original API, e.g., class or function, from which this documentation node is derived.
+	 */
+	readonly apiItemName: string;
+
+	/**
+	 * Category or type of the API like 'class' or 'function'.
+	 */
+	readonly apiItemKind: string;
+
+	/**
+	 * Originating package name for the API.
```

Comment:
Lets add a `@remarks` comment noting that documents corresponding to an entity that doesn't belong to a package (e.g. an `ApiModel`) will not have this field set.